presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestSimplifyCountOverInnerAggregateQuery.java

```diff
@@ -0,0 +1,136 @@
+                                        new Signature("count", AGGREGATE, parseTypeSignature(StandardTypes.BIGINT)),
+                                        Optional.empty())), ImmutableList.of(ImmutableList.of()), AggregationNode.Step.SINGLE,
+                                Optional.empty(), Optional.empty())
+                ).doesNotFire();
+    }
+    @Test
+    public void testFiresOnNestedCountAggregate()
+    {
+        PlanNodeIdAllocator planNodeIdAllocator = new PlanNodeIdAllocator();
+        tester.assertThat(new SimplifyCountOverInnerAggregateQuery())
+                .on(p ->
+                        new AggregationNode(planNodeIdAllocator.getNextId(),
+                                new AggregationNode(planNodeIdAllocator.getNextId(), p.tableScan(ImmutableList.of(), ImmutableMap.of()),
+                                        ImmutableMap.of(),
+                                        ImmutableList.of(ImmutableList.of()), AggregationNode.Step.SINGLE, Optional.empty(), Optional.empty()),
+                                ImmutableMap.of(p.symbol("count_1", BigintType.BIGINT), new AggregationNode.Aggregation(
+                                        new FunctionCall(QualifiedName.of("count"), ImmutableList.of()),
+                                        new Signature("count", AGGREGATE, parseTypeSignature(StandardTypes.BIGINT)),
+                                        Optional.empty())),
+                                ImmutableList.of(ImmutableList.of()),
+                                AggregationNode.Step.SINGLE,
+                                Optional.empty(),
+                                Optional.empty())
+                ).matches(PlanMatchPattern.values(ImmutableMap.of("count_1", 0)));
+    }
+    @Test
+    public void testDoesNotFireOnNestedNonCountAggregate()
+    {
+        PlanNodeIdAllocator planNodeIdAllocator = new PlanNodeIdAllocator();
+        tester.assertThat(new SimplifyCountOverInnerAggregateQuery())
+                .on(p -> {
+                    Symbol totalprice = p.symbol("totalprice", DOUBLE);
+                    AggregationNode innerAgg = new AggregationNode(planNodeIdAllocator.getNextId(),
+                            p.project(
+                                    Assignments.of(p.symbol("totalprice", DOUBLE), totalprice.toSymbolReference()),
+                                    p.tableScan(
+                                            new TableHandle(
+                                                    new ConnectorId("local"),
+                                                    new TpchTableHandle("local", "orders", TINY_SCALE_FACTOR)),
+                                            ImmutableList.of(totalprice),
+                                            ImmutableMap.of(totalprice, new TpchColumnHandle(totalprice.getName(), DOUBLE)))),
+                            ImmutableMap.of(p.symbol("sum_inner", DOUBLE),
+                                    new AggregationNode.Aggregation(
```

Comment:
why not to use `p.aggregation(...)` here and in all the places where you create a new aggregation node?